crates/core/tedge_write/src/api.rs

```diff
@@ -33,24 +33,30 @@ impl<'a> CopyOptions<'a> {
     ///
     /// Stdin and Stdout are UTF-8.
     pub fn copy(self) -> anyhow::Result<()> {
-        let output = self
-            .command()?
-            .output()
-            .context("Starting tedge-write process failed")?;
+        let mut command = self.command()?;
+
+        let output = command.output();
+
+        let program = command.get_program().to_string_lossy();
+        let output = output.with_context(|| format!("failed to start process '{program}'"))?;
 
         if !output.status.success() {
+            let stderr = String::from_utf8(output.stderr).expect("output should be utf-8");
+            let stderr = stderr.trim();
+            let exit_code = output.status.code().unwrap();
```

Comment:
Here, isn't it better to be defenesive and account for the `None` status code case as well, as there is the theoretical possibility of the `tedge-write` process getting killed?

Reply:
Hmm, I think that `tedge-write` process being terminated by a signal is very unlikely, as it is spawned and joined by tedge-agent, but you're right, it's better for that edge case to be properly handled as well.